src/server/transaction.h

```diff
@@ -433,6 +431,24 @@ class Transaction {
     }
   };
 
+  // Barrier akin to helio's BlockingCounter, but with phase counting and proper acquire semantics
+  // for polling work from other threads. And without keap allocation.
+  struct PhasedBarrier {
+    uint32_t Start(uint32_t count);  // Release: Store count, return current phase
```

Comment:
phase is like seqlock? 
add a comment that it corresponds to a specific hop in a transaction

Reply:
phase is seqlock, I added a comment in transaction.h that a phase is a hop

Reply:
not that I suggest doing it in this PR but did you consider merging phase and count into the same atomic_uint32?

Reply:
No need, because #2457 will get rid of it at all